File: Backend/SkillForge/SkillForge/Models/DTOs/Home/HomePageData.cs
using SkillForge.Models.DTOs.Article;
using SkillForge.Models.DTOs.Tag;
using SkillForge.Models.DTOs.User;

namespace SkillForge.Models.DTOs.Home;

public class HomePageData
{
    public List<ArticleCard> LatestArticles { get; set; }

    public List<TopArticleItem> TopArticles { get; set; }

    public List<UserLink> TopUsers { get; set; }

    public List<TagLink> TopTags { get; set; }
}

File: Frontend/skillforge-common/types/HomePageData.ts
import type ArticleCardType from "./ArticleCardType";
import type TagLink from "./TagLinkType";
import type TopArticleItemType from "./TopArticleItemType";
import type UserLink from "./UserLinkType";

export default interface HomePageData {
    TopArticles: TopArticleItemType[],
    TopUsers: UserLink[],
    TopTags: TagLink[],
    LatestArticles: ArticleCardType[],
}